projects/packages/publicize/src/class-publicize-base.php

```diff
@@ -497,8 +497,8 @@ public function get_profile_link( $service_name, $connection ) {
 			return 'https://instagram.com/' . $cmeta['connection_data']['meta']['username'];
 		}
 
-		if ( 'threads' === $service_name && isset( $connection['external_name'] ) ) {
-			return 'https://www.threads.net/@' . $connection['external_name'];
+		if ( 'threads' === $service_name && isset( $cmeta['external_name'] ) ) {
+			return 'https://www.threads.net/@' . $cmeta['external_name'];
```

Comment:
This change is to ensure it works on WPCOM sites where connection is not an array.